terraform-provider-constellation/internal/provider/attestation_data_source.go

```diff
@@ -87,6 +102,33 @@ func (d *AttestationDataSource) Schema(_ context.Context, _ datasource.SchemaReq
 	}
 }
 
+// ValidateConfig validates the configuration for the image data source.
+func (d *AttestationDataSource) ValidateConfig(ctx context.Context, req datasource.ValidateConfigRequest, resp *datasource.ValidateConfigResponse) {
+	var data AttestationDataSourceModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if !data.AttestationVariant.Equal(types.StringValue("azure-sev-snp")) && !data.MaaURL.IsNull() {
+		resp.Diagnostics.AddAttributeError(
+			path.Root("maa_url"),
+			"MAA URL can only be set for Azure SEV-SNP", "When using other attestation variants, it should not be set.",
+		)
+		return
+	}
```

Comment:
Does Terraform treat `"null"` the same as `null`?
In any case, I don't think it is required to return a hard failure if optional values are set for CSPs that don't require them.